src/runtime/manifest-ast-nodes.ts

```diff
@@ -43,6 +43,10 @@ export class BaseNode {
   location: SourceLocation;
 }
 
+export class BaseNodeWithRefinement extends BaseNode {
+    refinement: Refinement|null;
```

Comment:
I think we can use:
```
refinement?: Refinement
```
which would make the field optional (and remove the need for the `refinement: null` changes.